### ISSUE TITLE
Tiro Bangla has interchanged two ligatures হ্ণ হ্ন

### ISSUE THREAD

**Issue by madhavkiransodum** [TiroTypeworks/Indigo#30]
In bengali, হ্ + ণ = U+09B9 U+09CD U+09A3 should be
![image](https://user-images.githubusercontent.com/16919364/180649769-72ee35a3-4427-46bb-b639-5fce86f84642.png)
and হ্ + ন = U+09B9 U+09CD U+09A8 should be
![image](https://user-images.githubusercontent.com/16919364/180649785-946bb90f-daf1-4f6c-91c9-cfa22fe57c89.png)
but the font now shows the opposite.

Adobe Bengali, Galada and Atma fonts have got it right but others have got them wrong.
![image](https://user-images.githubusercontent.com/16919364/180649816-f747e53f-0db1-495c-9dd4-e0dce08943c8.png)

**Comment by madhavkiransodum** [comment]
A further proof is the list of ligatures here:

One can see that ন (U+09A8) always comes at the bottom for all the other consonants, whereas ণ (U+09A3) always comes at the back of the consonant.
![image](https://user-images.githubusercontent.com/16919364/180649862-5811fa50-f3b8-400d-b48f-631489204f91.png)

Adobe fonts have got it right. The designer for both Adobe and Tiro is the same, so this issue should be rectified.
![image](https://user-images.githubusercontent.com/16919364/180649877-92312b0b-22b6-48c0-80d1-1301e94a092b.png)

There is a host of well edited literature in bengali with this right usage.. like the following: viṣa-vahnim from Haridas Das's Gopala virudavali
![image](https://user-images.githubusercontent.com/16919364/180649905-1ba953f6-4cc1-44c9-a211-eeda7d0bf06b.png)

Another write example: gṛhṇāti (from Sajjana tosani a 1872 bengali magazine)
![image](https://user-images.githubusercontent.com/16919364/180649914-a934ffe4-5031-4216-be68-26948c24da75.png)

**Comment by tiroj** [comment]
Thank you for posting this issue. It is interesting to me that various fonts from different foundries are switching these forms: that suggests either that they are all using the same reference, or that there has been a history of variation in the use of these forms. I will investigate further.

**Comment by madhavkiransodum** [comment]
Agreed. It could also be an oversight since ণ and ন look so similar and sound similar as well. And for those who use bengali script for bengali language these differences don't matter (just as the differences between স, ষ and শ are only a formality), but for those who use bengali script for sanskrit language these small details can make a difference... 

Nowadays very few people use bengali for sanskrit -- so not unusual that this happens...

**Comment by tiroj** [comment]
Thank you. That is a helpful additional insight.

On the subject, I am very interested in learning more about Sanskrit in Bengali script. Our Bengali fonts use a set of conjunct ligatures that are based on attested Bengali language conjuncts plus a smallish set that our newspaper clients have requested over the years for transcribing foreign loan words. I have a well-documented set of Sanskrit conjuncts, which are implemented in our Tiro Devanagari Sanskrit font, but I have no idea how most of these might have been written in Bengali script: at present, the Tiro Bangla font will display many of them with an explicit virama mark, which doesn’t seem very elegant. I am thinking that in the manuscript tradition there must be examples of conjunct ligatures in Sanskrit words that are not attested in Bengali language, and hence not found in typefaces. I would very much like to start documenting such forms, with the ultimate goal of extending the Tiro Bangla ligature set, or making a dedicated Tiro Bangla Sanskrit font if there turn out to be significant differences in preferred style or forms.

**Comment by madhavkiransodum** [comment]
That would be laudable! Could you please attach here a list of the conjuncts that Tiro Bangla currently has? I will try to help you with the documentation by posting the other conjuncts that haven't been included yet from sanskrit literature in bengali script...

**Comment by tiroj** [comment]
[I will open a separate issue re. Bengali conjuncts.]

**Comment by tiroj** [comment]
Fiona Ross, Neelakash Kshetrimayum and I have looked into the question around the forms of হ্ণ and হ্ন, and some important sources support our assignment of shapes in Tiro Bangla and other fonts.

This is from Vidyasagar’s primer _Barna Parichay_ (1855), which standardised modern Bengali spelling:
<img width="533" alt="image" src="https://user-images.githubusercontent.com/5614500/181054916-1090abf5-788b-4d8e-99fa-592ceb73a8f9.png">

This pattern is confirmed for us by Dr Sukanta Chaudhuri and Dr Hanna-Ruth Thomoson from SOAS. It is also the pattern approved by our clients at Anandabazer Patrika and the Bengali editors at Harvard University Press.

So at least from the mid-19th Century this mapping of ligature forms is considered correct for Bengali language. If there is an older mapping that differs, or that is non-standardised and varied, that would be interesting to document. And if there is a different convention in the mapping of these ligatures in Sanskrit manuscripts in Bengali script, that would be especially interesting. But for now I am recording this issue as being by design.

**Comment by sridatta1** [comment]
I agree there needs to be further research on this.
As the two letters are written similarly and pronounced identically in spoken bengali.
I checked some sanskrit manuscripts
here it is cihna 
![image](https://user-images.githubusercontent.com/23263723/181164745-3a70da7b-6eef-49d1-b68a-2c13d465037d.png)
https://eap.bl.uk/archive-file/EAP921-4-2-38#?c=0&m=0&s=0&cv=56&xywh=1725%2C189%2C1560%2C1015
vahni
![image](https://user-images.githubusercontent.com/23263723/181164878-59fedded-c496-4e7f-b118-7bf411213b6e.png)
One need to  হ্ণ is written in manuscripts and handwritings
This would also apply to ক্ষ্ণ
![image](https://user-images.githubusercontent.com/23263723/181165273-7f2b03d0-dd2a-4924-a200-6e8d317d37f4.png)
 Here the lower form is like NNA rather than NA.
Many fonts have the lower part in  হ্ণ ক্ষ্ণ similar to ণ rather than ন
Kohinoor Bangla
![image](https://user-images.githubusercontent.com/23263723/181165555-34104761-524e-4d77-9c30-b00492241759.png)
Tulika Bengali
![image](https://user-images.githubusercontent.com/23263723/181165785-5802ea00-4214-47cc-a0af-8d09f75a5059.png)
Tatsam Bengali
![image](https://user-images.githubusercontent.com/23263723/181165962-077b1d5b-b6ce-4ff5-83a4-4ca0bf4298b0.png)

**Comment by madhavkiransodum** [comment]
True... will need to research more ... Is it due to editorial preferences?

A book published in 2021 has it the other way:
![image](https://user-images.githubusercontent.com/16919364/181169135-7ce1effe-ff01-40db-a28a-f33ae5ceff25.png)
https://www.amazon.in/Prashna-Chinha-%E0%A6%AA%E0%A7%8D%E0%A6%B0%E0%A6%B6%E0%A7%8D%E0%A6%A8%E0%A6%9A%E0%A6%BF%E0%A6%B9%E0%A7%8D%E0%A6%A8-Foujul-Houque/dp/1714396681

And so does Mahabharata published in 1862:
![image](https://user-images.githubusercontent.com/16919364/181169294-5ae931d6-5ad0-4a03-a03c-63e7303c9113.png)
![image](https://user-images.githubusercontent.com/16919364/181169469-55925d26-62a5-4a4f-9be1-9628d87968b3.png)

https://www.google.co.in/books/edition/Mah%C4%81bh%C4%81rata/VSxLAAAAcAAJ?hl=en&gbpv=0

**Comment by sridatta1** [comment]
> This is from Vidyasagar’s primer Barna Parichay (1855), which standardised modern Bengali spelling:

Can you the year in which this copy of the book was reprinted?
In this copy of Barṇaparicaya dated 1876
The lower component of হ্ণ hṇa and ক্ষ্ণ kṣṇa is similar to ণ ṇa
![image](https://user-images.githubusercontent.com/23263723/182034628-2f29ba81-f65d-44b6-a56d-ad40c5e01d49.png)
![image](https://user-images.githubusercontent.com/23263723/182034662-73b2e07b-0f2d-43fd-bbce-3f09ee157a63.png)
ক্ষ্ণ kṣṇa
![image](https://user-images.githubusercontent.com/23263723/182035737-9c8c3d84-6624-46e0-9d8d-2c6c57092c9b.png)

http://access.bl.uk/item/viewer/ark:/81055/vdc_100086181011.0x000001#?c=0&m=0&s=0&cv=13&xywh=-668%2C-1%2C4472%2C2702


In this copy of the book from 1864
The lower part of হ্ণ hṇa is similar to ণ
![image](https://user-images.githubusercontent.com/23263723/182034979-2b666f46-251e-462c-9de8-4888466f6bc8.png)
and hna is like হ্ণ and the lower part is similar to ন
![image](https://user-images.githubusercontent.com/23263723/182035085-d7f248f4-931e-4efa-a997-34710dc9581f.png)
http://access.bl.uk/item/viewer/ark:/81055/vdc_100062717373.0x000001#?c=0&m=0&s=0&cv=13&xywh=-638%2C0%2C4325%2C2612

In an other edition from 1863, there is hṇa, but no example for hna
http://access.bl.uk/item/viewer/ark:/81055/vdc_100063587586.0x000001#?c=0&m=0&s=0&cv=14&xywh=-572%2C-1%2C4229%2C2555
![image](https://user-images.githubusercontent.com/23263723/182035664-42c1566c-b90e-4daa-916d-3fb0ece679eb.png)

**Comment by sridatta1** [comment]
The two conjuncts in manuscripts
 হ্ন
![image](https://user-images.githubusercontent.com/23263723/182655672-cc199445-c1af-4eaf-9f27-bb58e3961b79.png)

 হ্ণ (horizontal and vertical conjuncts)
![image](https://user-images.githubusercontent.com/23263723/182655898-fa3ba284-bd5e-4ebd-a73b-1967e99f08ea.png)

**Comment by madhavkiransodum** [comment]
Śrīmad Bhāgavatam Published by Nitya Svarūpa Brahmacārī in 1911 has this:

![image](https://user-images.githubusercontent.com/16919364/183612286-5c2ffafc-dd96-4b4b-8340-f4eafd6ec48f.png)

**Comment by tiroj** [comment]
I conducted an informal poll of Bangla readers on Twitter:
https://twitter.com/TiroTypeworks/status/1554888207711367169

While there is evidence of some difference of opinion and hence reading preference, the majority of respondents (61.5%) identified this shape
<img width="543" alt="image" src="https://user-images.githubusercontent.com/5614500/189432611-2340afb9-cf0b-49c0-93dd-484e5e1f3b89.png">
as হ+ন and _not_ as হ+ণ.

As the illustrations in this thread indicate, disagreement about this is not something new.

At this time, I am not taking any action on this issue, in terms of the shaping of these conjuncts, or the shape of the default glyphs used for either. However, I have designed variant outlines for the /bHNna/ ligature glyph (and also for the /bKSsNna/ glyph), with shapes that are more obviously –Nna. I will enable these as Stylistic Set variants in the next update to the fonts, and if someone wanted to they could fork the fonts to make these the default forms.
<img width="549" alt="image" src="https://user-images.githubusercontent.com/5614500/189433421-e85e54f1-925b-49b2-af58-9f527b220d85.png">

**Comment by tiroj** [comment]
Now testable with `v160beta.yml` build path.
